src/plugins/discover/public/application/main/utils/fetch_all.ts

```diff
@@ -93,9 +95,20 @@ export function fetchAll(
             recordRawType,
           });
         }
+        /**
+         * The partial state for text based query languages is necessary in case the query has changed
+         * In the follow up useTextBasedQueryLanguage hook in this case new columns are added to AppState
+         * So the data table shows the new columns of the table. The partial state was introduced to prevent
+         * To frequent change of state causing the table to re-render to often, which causes race conditions
+         * So it takes too long, a bad user experience, also a potential flakniess in tests
+         */
+        const fetchStatus =
+          useSql && (!prevQuery || !isEqual(query, prevQuery))
+            ? FetchStatus.PARTIAL
+            : FetchStatus.COMPLETE;
```

Comment:
this is the main change in this PR, text based query lang results are given a PARTIAL state in chase the query has changed, because the result might have change Discover's appState (.e.g. adding default columns). The UI considers the PARTIAL state as LOADING and an unnecessary re-rendering of the expensive EuiDataGrid rendering is omitted.